gravitee-apim-console-webui/src/management/application/details/subscriptions/creation/application-subscription-creation-dialog.component.ts

```diff
@@ -0,0 +1,247 @@
+import { Observable, of, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged, filter, map, share, switchMap, tap } from 'rxjs/operators';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { GioJsonSchema } from '@gravitee/ui-particles-angular';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import { toNewSubscriptionEntity } from './application-subscription-creation-dialog.adapter';
+import {
+  ApplicationSubscriptionCreationDialogData,
+  ApplicationSubscriptionCreationDialogResult,
+  SubscriptionCreationForm,
+} from './application-subscription-creation-dialog.models';
+
+import { ApiV2Service } from '../../../../../services-ngx/api-v2.service';
+import { Api, Plan, PlanV4 } from '../../../../../entities/management-api-v2';
+import { ApiPlanV2Service } from '../../../../../services-ngx/api-plan-v2.service';
+import { ApplicationService } from '../../../../../services-ngx/application.service';
+import { EnvironmentSettingsService } from '../../../../../services-ngx/environment-settings.service';
+import { ApiKeyMode, Application } from '../../../../../entities/application/Application';
+import { ApplicationSubscription } from '../../../../../entities/subscription/subscription';
+import { ConnectorPluginsV2Service } from '../../../../../services-ngx/connector-plugins-v2.service';
+import { IconService } from '../../../../../services-ngx/icon.service';
+
+@Component({
+  selector: 'application-subscription-creation-dialog',
+  templateUrl: './application-subscription-creation-dialog.component.html',
+  styleUrls: ['./application-subscription-creation-dialog.component.scss'],
+})
+export class ApplicationSubscriptionCreationDialogComponent {
+  private destroyRef = inject(DestroyRef);
+  private canUseCustomApiKey = this.environmentSettingsService.getSnapshot().plan?.security?.customApiKey?.enabled;
+  private canUseSharedApiKeys = this.environmentSettingsService.getSnapshot().plan?.security?.sharedApiKey?.enabled;
+  private apiKeyModeSubscription: Subscription;
+  private apiKeySubscriptions: ApplicationSubscription[];
+  private entrypointsMap = new Map<string, { icon: string; name: string }>();
+  protected availableSubscriptionEntrypoints: { type: string; name?: string; icon?: string }[] = [];
+  protected application: Application;
+  protected selectedSchema: GioJsonSchema;
+  protected form: SubscriptionCreationForm = new FormGroup({
+    selectedApi: new FormControl(null),
+    selectedPlan: new FormControl(null, [Validators.required]),
+  });
+  protected plans$: Observable<Plan[]>;
+  protected apis$: Observable<Api[]> = this.form.controls.selectedApi.valueChanges.pipe(
+    distinctUntilChanged(),
+    debounceTime(100),
+    filter((term) => typeof term === 'string'),
+    switchMap((term: string) => (term.length > 0 ? this.apiService.search({ query: term }) : of(null))),
```

Comment:
Could you check if we need to add `manageOnly=false`.
Because this is an api search from an application point of view. 
So we can subscribe to an api that we can't see in the console (but in the portal yes)